system-tests/tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/TracingIntegrationTests.java

```diff
@@ -0,0 +1,138 @@
+            "TransferProcessManagerImpl.processInitial",
+            "TransferProcessManagerImpl.processProvisioned",
+            "TransferProcessManagerImpl.initiateProviderRequest"
+    };
+
+    @BeforeAll
+    static void startGrpcServer() {
+        grpcServer = new OtlpGrpcServer();
+        grpcServer.start();
+    }
+
+    @AfterAll
+    static void stopGrpcServer() {
+        grpcServer.stop().join();
+    }
+
+    @BeforeEach
+    void resetGrpcServer() {
+        grpcServer.reset();
+        GlobalOpenTelemetry.resetForTest();
+    }
+
+    @Test
+    void transferFile_testTraces() throws Exception {
+        // Arrange
+        // Create a file with test data on provider file system.
+        String fileContent = "FileTransfer-test-" + UUID.randomUUID();
+        Files.write(Path.of(PROVIDER_ASSET_PATH), fileContent.getBytes(StandardCharsets.UTF_8));
+
+        // Act
+        runGatling(FileTransferLocalSimulation.class, FileTransferSimulationUtils.DESCRIPTION);
+
+        // Assert
+        await().atMost(30, SECONDS).untilAsserted(() -> {
+                    // Get exported spans.
+                    List<Span> spans = grpcServer
+                            .traceRequests
+                            .stream()
+                            .flatMap(r -> r.getResourceSpansList().stream())
+                            .flatMap(r -> r.getInstrumentationLibrarySpansList().stream())
+                            .flatMap(r -> r.getSpansList().stream())
+                            .collect(Collectors.toList());
+
+                    // Assert that expected spans are present.
+                    List<Span> contractNegotiationSpans = getSpans(spans, Arrays.stream(contractNegotiationSpanNames));
+                    List<Span> transferProcessSpans = getSpans(spans, Arrays.stream(transferProcessSpanNames));
+
+                    for (Span span: contractNegotiationSpans) {
+                        System.out.println(span.getName() + " " + span.getTraceId());
+                    }
+
+                    System.out.println(" ");
+                    for (Span span: transferProcessSpans) {
+                        System.out.println(span.getName() + " " + span.getTraceId());
+                    }
+                    // Assert that spans are part of the right trace.
+                    assertSpansHaveSameTrace(contractNegotiationSpans);
+                    assertSpansHaveSameTrace(transferProcessSpans);
+                }
+        );
+    }
+
+    private List<Span> getSpans(List<Span> spans, Stream<String> spanNames) {
+        return spanNames.map(spanName -> getSpanByName(spans, spanName)).collect(Collectors.toList());
+    }
+
+    private void assertSpansHaveSameTrace(List<Span> spans) {
+        assertThat(spans.stream().map(s -> s.getTraceId().toStringUtf8()).distinct())
+                .withFailMessage(() -> "Spans from the same trace should have the same traceId.")
+                .singleElement();
```

Comment:
`singleElement()` seems unnecessary

Reply:
It is necessary to assert that there is a single element. The IDE highlights it because the result is never used later but it is actually used for the assertion.